src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/SynthesizedSymbols/AnonymousType.TemplateSymbol.cs

```diff
@@ -133,7 +133,7 @@ internal AnonymousTypeTemplateSymbol(AnonymousTypeManager manager, AnonymousType
 
             internal AnonymousTypeKey GetAnonymousTypeKey()
             {
-                var properties = this.Properties.SelectAsArray(p => AnonymousTypeKeyField.CreateField(p.Name));
+                var properties = this.Properties.SelectAsArray(p => new AnonymousTypeKeyField(p.Name, isKey: false, ignoreCase: false));
```

Comment:
AnonymousTypeKeyField did convert the name to lower-case (when it was created from VB) instead of comparing case-insensitively. Lowering case and then comparing ordinal is incorrect.